packages/oauth-helper/src/index.ts

```diff
@@ -0,0 +1,468 @@
+          bot: {
+            scopes: resp.scope.split(','),
+            token: resp.access_token,
+            userId: resp.bot_user_id,
+            id: '', // not currently included
+          },
+          tokenType: resp.token_type,
+        };
+      }
+
+      // save access code to installationStore
+      await this.installationStore.storeInstallation(installation);
+      if (options != null && options.success != null) {
+        console.log('calling options.sucess');
+        options.success(installation, metadata, req, res);
+      } else {
+        // call our default callbackSuccess function
+        callbackSuccess(installation, metadata, req, res);
+      }
+    } catch (error) {
+      console.log(error);
+      if (options != null && options.failure != null) {
+        console.log('calling options.failure');
+        options.failure(error, this.installOptions , req, res);
+      } else {
+        console.log('run custom failure function');
+        callbackFailure(error, this.installOptions , req, res);
+      }
+    }
+  }
+}
+
+interface InstallProviderOptions {
+  clientId: string;
+  clientSecret: string;
+  stateStore?: StateStore; // default ClearStateStore
+  stateSecret?: string; // ClearStateStoreOptions['secret']; // required when using default stateStore
+  installationStore?: InstallationStore; // default MemoryInstallStore
+  authVersion: 'v1' | 'v2'; // default 'v2'
+}
+
+interface InstallURLOptions {
+  scopes: string | string[];
+  teamId?: string;
+  redirectUri?: string;
+  userScopes?: string | string[]; // cannot be used with authVersion=v1
+  metadata?: string;
+}
+
+interface CallbackOptions {
+  // success is given control after handleCallback() has stored the
+  // installation. when provided, this function must complete the
+  // callbackRes.
+  success?: (
+    installation: Installation,
+    metadata: string | undefined,
+    callbackReq: IncomingMessage,
+    callbackRes: ServerResponse,
+  ) => void;
+
+  // failure is given control when handleCallback() fails at any point.
+  // when provided, this function must complete the callbackRes.
+  // default:
+  // serve a generic "Error" web page (show detailed cause in development)
+  failure?: (
+    error: CodedError,
+    options: InstallURLOptions,
+    callbackReq: IncomingMessage,
+    callbackRes: ServerResponse,
+  ) => void;
+}
+
+// Response shape from oauth.v2.access - https://api.slack.com/methods/oauth.v2.access#response
+interface OAuthV2Response {
+  ok: boolean;
+  app_id: string;
+  authed_user: {
+    id: string,
+    scope?: string,
+    access_token?: string,
+    token_type?: string,
+  };
+  scope: string;
+  token_type: string;
+  access_token: string;
+  bot_user_id: string;
+  team: { id: string, name: string };
+  enterprise: { name: string, id: string } | null;
+  response_metadata: object;
+}
+
+// Response shape from oauth.access - https://api.slack.com/methods/oauth.access#response
+interface OAuthV1Response {
```

Comment:
purposely didn't include workspace apps response values in here. @aoberoi said we shouldn't bother supporting workspace apps since they are deprecated